File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/handshake/ProtocolRequired.scala





















// Generated on 05/03/2013 10:35:00
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait ProtocolRequired extends MessageDeserializer {
  val messageId = 1

  protected trait Message extends BaseMessage with Serializable {
    val messageId = ProtocolRequired.messageId

    val requiredVersion: Int
    val currentVersion: Int

    def serialize(buf: Buffer) {
      buf.writeInt(requiredVersion)
      buf.writeInt(currentVersion)
    }

  }

  protected def create(requiredVersion: Int, currentVersion: Int): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val requiredVersion = buf.readInt()
    val currentVersion = buf.readInt()

    create(requiredVersion, currentVersion)
  }

}

object ProtocolRequired extends ProtocolRequired {
  case class MessageImpl(requiredVersion: Int, currentVersion: Int) extends Message
  type Target = MessageImpl

  override def create(requiredVersion: Int, currentVersion: Int) = MessageImpl(requiredVersion, currentVersion)

  def apply(requiredVersion: Int, currentVersion: Int) = create(requiredVersion, currentVersion)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/ProtocolRequired.java


// Generated on 05/08/2013 19:37:58
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class ProtocolRequired extends NetworkMessage {
    public static final int MESSAGE_ID = 1;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public int requiredVersion;
    public int currentVersion;
    
    public ProtocolRequired() { }
    
    public ProtocolRequired(int requiredVersion, int currentVersion) {
        this.requiredVersion = requiredVersion;
        this.currentVersion = currentVersion;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeInt(requiredVersion);
        buf.writeInt(currentVersion);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        requiredVersion = buf.readInt();
        if (requiredVersion < 0)
            throw new RuntimeException("Forbidden value on requiredVersion = " + requiredVersion + ", it doesn't respect the following condition : requiredVersion < 0");
        currentVersion = buf.readInt();
        if (currentVersion < 0)
            throw new RuntimeException("Forbidden value on currentVersion = " + currentVersion + ", it doesn't respect the following condition : currentVersion < 0");
    }
    
}
